File: 2_using_communicators/solutions/comm.c
#include <stdio.h>
#include <mpi.h>

void main(int argc, char * argv[]) {
     int err;
     int nprocs, my_rank;
     err = MPI_Init(&argc, &argv);
     MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
     MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
     printf("Hello I am %d of %d procs \n", my_rank, nprocs);
     err = MPI_Finalize();
}

File: 2_using_communicators/solutions/comm.f90
program my_prog
   use mpi
   implicit none
   integer :: ierr, nprocs, my_rank
   call mpi_init(ierr)
   call mpi_comm_size(mpi_comm_world, nprocs, ierr)
   call mpi_comm_rank(mpi_comm_world, my_rank, ierr)
   write(*,*) "Hello I am ",my_rank," of ", nprocs
   call mpi_finalize(ierr)
end program my_prog
